packages/appeals-service-api/src/repositories/sql/service-user-repository.js

```diff
@@ -26,6 +38,88 @@ class ServiceUserRepository {
 			}
 		});
 	}
+
+	/**
+	 * @param {Omit<ServiceUser, 'internalId'>} data
+	 * @returns
+	 */
+	async put(data) {
+		return this.dbClient.$transaction(async (tx) => {
+			/** @type {import('@prisma/client').AppealUser | null} */
+			let appealUser = null;
+			if (data.emailAddress) {
+				appealUser = await tx.appealUser.findFirst({
+					where: {
+						email: data.emailAddress
+					}
+				});
+
+				if (!appealUser) {
+					appealUser = await tx.appealUser.create({
+						data: {
+							email: data.emailAddress,
+							serviceUserId: data.id
+						}
+					});
+				} else {
+					appealUser = await tx.appealUser.update({
+						where: {
+							email: data.emailAddress
+						},
+						data: {
+							serviceUserId: data.id
+						}
+					});
+				}
+			}
+
+			const role = mapDataModelRoleToInternalRole(data.serviceUserType);
+			if (role && appealUser) {
+				const appealCase = await tx.appealCase.findFirst({
+					where: {
+						caseReference: data.caseReference
+					},
+					select: {
+						Appeal: {
+							select: {
+								id: true
+							}
+						}
+					}
+				});
+
+				if (appealCase) {
+					await tx.appealToUser.create({
```

Comment:
am I right in thinking this will create a duplicate appealToUser record if the service user is updated and this already exists? thinking it needs to be an upsert on appealUser.id

Reply:
ah yep good point, will change